### ISSUE TITLE
"detect error" terminal output every time even face is available in the video feed

### ISSUE THREAD

**Issue by tomriddle54** [TreB1eN/InsightFace_Pytorch#79]
@TreB1eN 
when I ran `face_verify.py` removed try catch to debug
got this error 
`traceback (most recent call last):
  File "face_verify.py", line 55, in <module>
    bboxes, faces = mtcnn.align_multi(image, conf.face_limit, conf.min_face_size)
`

`anaconda3/envs/pytorch-face/lib/python3.7/site-packages/numpy/core/shape_base.py", line 234, in vstack
    return _nx.concatenate([atleast_2d(_m) for _m in tup], 0)
ValueError: need at least one array to concatenate`
but faces are available in the video its not detecting properly for face nearly 80 percent time face is not detecting 20 percent time its detecting when the face is not detecting i am getting the above error or this `detect error`
Is there any way to make face detection strong (or maybe I am doing someting wrong I have followed all the instructions but there were lot of errors so i have solved it may be due to that )
I have provide same video to opencv's haar cascade its detecting face when I ran `face_verify.py `its not even say unknow terminal out put is `detect error`

**Comment by TreB1eN** [comment]
do you have image saved in the face bank ?

**Comment by tomriddle54** [comment]
Yes I have saved it in facebank the error is it's not detecting face

**Comment by engmubarak48** [comment]
> Yes I have saved it in facebank the error is it's not detecting face

Having the same problem, I think the MTCNN is not detecting faces as expected. even if there are no many images in the facebank, it should give 'UNKNOWN FACE'. 
@TreB1eN  How can we improve the performance of the detection part (MTCNN)? 
Is there some kind of threshold, that we can tune. ?

**Comment by engmubarak48** [comment]
@syedmustafa54 @TreB1eN 

#56 solves the problem

**Comment by tomriddle54** [comment]
@engmubarak48 thanks now its detecting face but what is the recognition algorithm its knn or something else

**Comment by engmubarak48** [comment]
> @engmubarak48 thanks now its detecting face but what is the recognition algorithm its knn or something else

@syedmustafa54 I also have the false positive issue, I think there is some threshold to tune. I will read the code and check. 

About the recognition, please read Insightface repo. I don't think there is something related to KNN.

**Comment by engmubarak48** [comment]
@syedmustafa54 

There is 'threshold to decide identical faces' in face_verify.py script, can you try to tune this parameter. it might improve your performance.

**Comment by tomriddle54** [comment]
> > @engmubarak48 thanks now its detecting face but what is the recognition algorithm its knn or something else
> 
> @syedmustafa54 I also have the false positive issue, I think there is some threshold to tune. I will read the code and check.
> 
> About the recognition, please read Insightface repo. I don't think there is something related to KNN.

solved it by controlling threshold

**Comment by engmubarak48** [comment]
> > > @engmubarak48 thanks now its detecting face but what is the recognition algorithm its knn or something else
> > 
> > 
> > @syedmustafa54 I also have the false positive issue, I think there is some threshold to tune. I will read the code and check.
> > About the recognition, please read Insightface repo. I don't think there is something related to KNN.
> 
> solved it by controlling threshold

happy you solved your problem. which threshold value worked for you.?

**Comment by tomriddle54** [comment]
@engmubarak48 in Insightface repo  there is nothing about recognition code

**Comment by tomriddle54** [comment]
> > > > @engmubarak48 thanks now its detecting face but what is the recognition algorithm its knn or something else
> > > 
> > > 
> > > @syedmustafa54 I also have the false positive issue, I think there is some threshold to tune. I will read the code and check.
> > > About the recognition, please read Insightface repo. I don't think there is something related to KNN.
> > 
> > 
> > solved it by controlling threshold
> 
> happy you solved your problem. which threshold value worked for you.?

0.9 threshold value worked for me its better if we chat and share our knowledge here is my discord channel create for face https://discord.gg/WMkzTP lets discuss there

**Comment by ZZH2950228** [comment]
names = np.load(conf.facebank_path/'names.npy')   there is a names.npy in this code , and what is it. If it is the idx list ,what is it format.
